test/types/schema.test.ts

```diff
@@ -318,4 +318,191 @@ function gh10900(): void {
   const patientSchema = new Schema<IUserProp>({
     menuStatus: { type: Schema.Types.Mixed, default: {} }
   });
-}
\ No newline at end of file
+}
+
+export function autoTypedSchema() {
+  // Test auto schema type obtaining with all possible path types.
+
+  class Int8 extends SchemaType {
+    constructor(key, options) {
+      super(key, options, 'Int8');
+    }
+    cast(val) {
+      let _val = Number(val);
+      if (isNaN(_val)) {
+        throw new Error('Int8: ' + val + ' is not a number');
+      }
+      _val = Math.round(_val);
+      if (_val < -0x80 || _val > 0x7F) {
+        throw new Error('Int8: ' + val +
+          ' is outside of the range of valid 8-bit ints');
+      }
+      return _val;
+    }
+  }
+
+  type TestSchemaType = {
+    string1?: string;
+    string2?: string;
+    string3?: string;
+    string4?: string;
+    number1?: number;
+    number2?: number;
+    number3?: number;
+    number4?: number;
+    date1?: Date;
+    date2?: Date;
+    date3?: Date;
+    date4?: Date;
+    buffer1?: Buffer;
+    buffer2?: Buffer;
+    buffer3?: Buffer;
+    buffer4?: Buffer;
+    boolean1?: boolean;
+    boolean2?: boolean;
+    boolean3?: boolean;
+    boolean4?: boolean;
+    mixed1?: any;
+    mixed2?: any;
+    mixed3?: any;
+    objectId1?: Schema.Types.ObjectId;
+    objectId2?: Schema.Types.ObjectId;
+    objectId3?: Schema.Types.ObjectId;
+    customSchema?: Int8;
+    map1?: Map<string, string>;
+    map2?: Map<string, number>;
+    array1?: string[];
+    array2?: any[];
+    array3?: any[];
+    array4?: any[];
+    array5?: any[];
+  };
+
+  const TestSchema = new Schema({
```

Comment:
We are missing Schema.Types.Decimal128

`
    decimal1: Schema.Types.Decimal128,`